xray/default_emitter.go

```diff
@@ -98,17 +100,22 @@ func packSegments(seg *Segment, outSegments [][]byte) [][]byte {
 			cb := ss.StreamCompletedSubsegments(s)
 			outSegments = append(outSegments, cb...)
 		}
-		b, _ := json.Marshal(s)
+		b, err := json.Marshal(s)
+		if err != nil {
+			logger.Errorf("JSON error while marshalling (Sub)Segment: %v", err)
```

Comment:
If there is an error, the program will panic right? Why not we panic at here?

Reply:
Current behavior - If there is error, b will be nil, and we don't stream that (Sub)Segment. 
So logging the error helps if there is any error. If we panic instead of error, this is "behavioral" change.